src/types/st-object.ts

```diff
@@ -56,6 +113,19 @@ class STObject extends SerializedType {
     const list: BytesList = new BytesList();
     const bytes: BinarySerializer = new BinarySerializer(list);
 
+    Object.keys(value).forEach((f) => {
+      if (
```

Comment:
Can you comment this condition - it's not obvious what is happening?

Reply:
`Field[f] !== undefined` : so you don't call `name` of undefined 
`type.name` denotes that this is of type `AccountID` 
`/^X/.test(value[f])` tests to see if the value associated with the field starts w/ X and should be treated an an x address.